pkg/client/restclient/config.go

```diff
@@ -158,6 +163,12 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 	if config.NegotiatedSerializer == nil {
 		return nil, fmt.Errorf("NegotiatedSerializer is required when initializing a RESTClient")
 	}
+	if config.QPS == 0.0 {
```

Comment:
> could you copy them to two local variables and pass them to NewRESTClient?

This doesn't conflict with my recommendation to put this defaulting logic here. :) Not modifying config seems good.